inference-engine/ie_bridges/c/samples/object_detection_sample_ssd/main.c

```diff
@@ -344,13 +360,17 @@ int main(int argc, char **argv) {
     }
     // -----------------------------------------------------------------------------------------------------
 
-    // --------------------------- 4. Read IR Generated by ModelOptimizer (.xml and .bin files) ------------
-    input_weight = (char *)calloc(strlen(input_model) + 1, sizeof(char));
-    memcpy(input_weight, input_model, strlen(input_model) - 4);
-    memcpy(input_weight + strlen(input_model) - 4, ".bin", strlen(".bin") + 1);
+    // 4. Read a model in OpenVINO Intermediate Representation (.xml and .bin files) or ONNX (.onnx file) format
     printf("%sLoading network files:\n", info);
     printf("\t%s\n", input_model);
-    printf("\t%s\n", input_weight);
+
+    if (getFileNameExt(input_model) == "xml") {
+        //read .bin weights for IR format only
+        input_weight = (char*)calloc(strlen(input_model) + 1, sizeof(char));
+        memcpy(input_weight, input_model, strlen(input_model) - 4);
+        memcpy(input_weight + strlen(input_model) - 4, ".bin", strlen(".bin") + 1);
```

Comment:
Why do we need this code? If xml and bin files have the same names we can pass only name of xml file.

Reply:
Yes, we can, we do that in all other samples, but I think it is better to leave this code to show users how they can manually load weight files. Are we going to deprecate this functionality at all?

Reply:
Fixed